byway/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.LoginState;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private static final String URL_TO_REDIRECT_TO_AFTER_LOGSIN = "/home.html";
+  private static final String URL_TO_REDIRECT_TO_AFTER_LOGSOUT = "/index.html";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    boolean loginStatus = userService.isUserLoggedIn();
+    String url;
+    if (!loginStatus) {
```

Comment:
Is this logic backwards? If login status is true, you want to redirect to a page where they log out? Or is the idea to provide a link for the opposite state of the current user (e.g. logged in users get a log out link, and not logged in users see a link to log in)?  

Additionally, if you have a simple split if/else in logic like this with a boolean value, put the true case first. It makes it easier to read and reason about.

Reply:
we plan to have a logout button in the header of the rest of the pages, so for now I changed it so that if user is logged in it displays a link to go to the landing page, and if they are not logged in, it displays a login button.